rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/NodeSourceWindow.java

```diff
@@ -0,0 +1,452 @@
+                                                                                                nodesRecoverableItem,
+                                                                                                buttonList,
+                                                                                                isNodeSourceEdited()));
+        cancelButton.addClickHandler(clickEvent -> window.hide());
+        buttonsLayout.setMembers(createAndDeployNodeSourceButton, createOnlyNodeSourceButton, cancelButton);
+
+        VLayout scrollLayout = new VLayout();
+        scrollLayout.setHeight100();
+        scrollLayout.setWidth100();
+        scrollLayout.setMembers(nodeSourcePluginsLayout);
+        scrollLayout.setOverflow(Overflow.AUTO);
+        scrollLayout.setBorder("1px solid #ddd");
+        scrollLayout.setBackgroundColor("#fafafa");
+
+        nodeSourceWindowLayout.addMember(nodeSourceWindowLabel);
+        nodeSourceWindowLayout.addMember(nodeSourceWindowForm);
+        nodeSourceWindowLayout.addMember(scrollLayout);
+        nodeSourceWindowLayout.addMember(buttonsLayout);
+
+        int winWidth = com.google.gwt.user.client.Window.getClientWidth() * 80 / 100;
+        int winHeight = com.google.gwt.user.client.Window.getClientHeight() * 80 / 100;
+        winWidth = Math.min(1000, winWidth);
+        winHeight = Math.min(1000, winHeight);
+
+        this.window = new Window();
+        this.window.setTitle(this.windowTitle);
+        this.window.setShowMinimizeButton(false);
+        this.window.setIsModal(true);
+        this.window.setShowModalMask(true);
+        this.window.addItem(nodeSourceWindowLayout);
+        this.window.setWidth(winWidth);
+        this.window.setHeight(winHeight);
+        this.window.setCanDragResize(true);
+        this.window.setCanDragReposition(true);
+        this.window.centerInPage();
+    }
+
+    private void prepareCreateAndDeployFormAndSubmit(VLayout nodeSourceWindowLayout,
+            Label nodeSourcePluginsWaitingLabel, DynamicForm nodeSourcePluginsForm, Label nodeSourceWindowLabel,
+            TextItem nodeSourceNameItem, CheckboxItem nodesRecoverableItem, List<IButton> buttonList,
+            boolean nodeSourceEdited) {
+
+        nodeSourcePluginsForm.setValue("deploy", Boolean.TRUE.toString());
+
+        prepareCreateOnlyFormAndSubmit(nodeSourceWindowLayout,
+                                       nodeSourcePluginsWaitingLabel,
+                                       nodeSourcePluginsForm,
+                                       nodeSourceWindowLabel,
+                                       nodeSourceNameItem,
+                                       nodesRecoverableItem,
+                                       buttonList,
+                                       nodeSourceEdited);
+    }
+
+    private void prepareCreateOnlyFormAndSubmit(VLayout nodeSourceWindowLayout, Label nodeSourcePluginsWaitingLabel,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Method has 8 parameters, which is greater than 7 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00107)